src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -318,6 +358,13 @@ protected override void OnEventCommand(EventCommandEventArgs command)
             }
         }
 
+        private static string NewLineSeparate(string str1, string str2)
+        {
+            if (string.IsNullOrEmpty(str1))
+                return str2;
```

Comment:
NewLineSeparate is small helper so that the callers are a bit shorter.    Those code sites never pass null.   I don't mind asserting that fact...